sherpa/utils/__init__.py

```diff
@@ -1780,30 +1780,30 @@ def multinormal_pdf(x, mu, sigma):
     .. [1] http://en.wikipedia.org/wiki/Multivariate_normal_distribution
 
     """
-    x = numpy.asarray(x)
-    mu = numpy.asarray(mu)
-    sigma = numpy.asarray(sigma)
+    x = np.asarray(x)
+    mu = np.asarray(mu)
+    sigma = np.asarray(sigma)
     if x.size != mu.size:
         raise TypeError("x and mu sizes do not match")
     if mu.size != sigma.diagonal().size:
         raise TypeError("sigma shape does not match x")
-    if numpy.min(numpy.linalg.eigvalsh(sigma)) <= 0:
+    if np.min(np.linalg.eigvalsh(sigma)) <= 0:
         raise ValueError("sigma is not positive definite")
-    if numpy.max(numpy.abs(sigma - sigma.T)) >= 1.e-9:
+    if np.max(np.abs(sigma - sigma.T)) >= 1.e-9:
         raise ValueError("sigma is not symmetric")
     rank = mu.size
-    coeff = 1.0 / (numpy.power(2.0 * numpy.pi, rank / 2.0) *
-                   numpy.sqrt(numpy.abs(numpy.linalg.det(sigma))))
-    xmu = numpy.mat(x - mu)
-    invsigma = numpy.mat(numpy.linalg.inv(sigma))
+    coeff = 1.0 / (np.power(2.0 * np.pi, rank / 2.0) *
+                   np.sqrt(np.abs(np.linalg.det(sigma))))
+    xmu = np.asmatrix(x - mu)
+    invsigma = np.asmatrix(np.linalg.inv(sigma))
```

Comment:
matrix itself is deprecated. While it's not removed right now in 2.0, it will be soon (https://numpy.org/doc/stable/reference/generated/numpy.matrix.html#numpy.matrix). Instead, one should just use normal ndarrays, that offer all the same functionality, just with different  indexing.
Do we want to do that right now or in the separate PR?

Reply:
I would rather do it in a separate PR because

 1. I don't have time at the moment
 2. this change needs some work (it's not just as simple as changing `asmatrix` to `asarray`, since you have to look at the code to see what it's doing, even if at the end of the day we do just go with the name change)
 3. there are likely other places in the code that `asmatrix` / `matrix` are used, not just the ones that `ruff` found were using `np.mat`

Reply:
Sounds goo to me. I'll open a new issue to keep track of this, but it's not needed for numpy 2.0.

Reply:
see #2026